File: backend-java/src/main/java/associados/api/mapper/AssociadoViewMapper.java
package associados.api.mapper;

import associados.api.dto.AssociadoView;
import associados.api.model.Associado;
import org.springframework.stereotype.Component;

@Component
public class AssociadoViewMapper implements Mapper <Associado, AssociadoView> {
    @Override
    public AssociadoView map(Associado associado) {
        return new AssociadoView(
                associado.getId(),
                associado.getNome(),
                associado.getEndereco(),
                associado.getBairro(),
                associado.getCidade(),
                associado.getEstado(),
                associado.getMesPago(),
                associado.getAnoPago()
        );
    }
}

File: backend/src/main/kotlin/associados/api/mapper/AssociadoViewMapper.kt
package associados.api.mapper

import associados.api.dto.AssociadoView
import associados.api.model.Associado
import org.springframework.stereotype.Component

@Component
class AssociadoViewMapper: Mapper<Associado, AssociadoView> {
    override fun map(t: Associado): AssociadoView {
        return AssociadoView(
            id = t.id,
            nome = t.nome
        )
    }
}
